autorest/codegen/models/parameter_list.py

```diff
@@ -145,6 +145,12 @@ def build_flattened_object(self) -> str:
             lambda parameter: parameter.location == ParameterLocation.Other
             and not isinstance(parameter.schema, ConstantSchema)
         )
-        parameter_string = ", ".join([f"{param.serialized_name}={param.serialized_name}" for param in parameters])
         object_schema = cast(ObjectSchema, self.body.schema)
+        parameter_string = ", ".join(
+            [f"{model_property.name}={param.serialized_name}"
+            for param, model_property in zip(
```

Comment:
Each parameter has a targetProperty that will avoid a zip if condition:
![image](https://user-images.githubusercontent.com/1050156/74386087-f6951a80-4da9-11ea-8dd9-4aec5433e6ca.png)